src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/HearingPartService.java

```diff
@@ -36,21 +46,59 @@ public HearingPart save(HearingPart hearingPart) {
         return hearingPartRepository.save(hearingPart);
     }
 
-    public HearingPart assignHearingPartToSession(UUID hearingPartId,
+    @Transactional
+    public UserTransaction saveWithTransaction(HearingPart hearingPart, UUID transactionId) {
+        HearingPart savedHearingPart = hearingPartRepository.save(hearingPart);
+
+        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
+        userTransactionDataList.add(new UserTransactionData("hearingPart",
+                savedHearingPart.getId(),
+                null,
+                "update",
+                "update",
+                0)
+        );
```

Comment:
2 sessions should be here as well as these are form business point of few also parts of the transaction